### ISSUE TITLE
How to train the model

### ISSUE THREAD

**Issue by shuuchen** [facebookresearch/PoseWarper#18]
Hi,

I notice that you used torch.no_grad() in the stem net, with no regard to train or inference.
Is the stem net parameter-fixed all the time ? or you trained it at somewhere else ?

Thank you !

**Comment by gberta** [comment]
Initially, the main backbone is trained in a single-frame setting. Afterwards, the warping offsets are learned in a multi-frame setting while the backbone is frozen.

**Comment by ArchNew** [comment]
> Initially, the main backbone is trained in a single-frame setting. Afterwards, the warping offsets are learned in a multi-frame setting while the backbone is frozen.

So, there's no joint training of two HRNet and the PoseWarper model?

**Comment by gberta** [comment]
No. We observed that joint training was detrimental to the pose detection accuracy. In other words, during joint training the model focuses more on correctly warping the keypoints from one frame to another, while the pose detection accuracy drops. If we do separate training, are able to maintain a good pose detection accuracy, and also good warping accuracy without sacrificing one for another.
